toonz/sources/tnztools/toolhandle.cpp

```diff
@@ -124,12 +124,17 @@ void ToolHandle::onImageChanged(TImage::Type imageType) {
   case TImage::TOONZ_RASTER:
     targetType = TTool::ToonzImage;
     break;
+  case TImage::MESH:
+    targetType = TTool::MeshImage;
+    break;
+  case TImage::META:
+    targetType = TTool::MetaImage;
+    break;
   case TImage::VECTOR:
-  default:
     targetType = TTool::VectorImage;
     break;
-  case TImage::MESH:
-    targetType = TTool::MeshImage;
+  default:
```

Comment:
Changing the default targetType for unknown image types to EmptyTarget is causing the Plastic tool to not work on Sub Xsheets.

Before the default target type was VectorImage, which the Plastic tool is configured for.  I do not recommend changing Plastic tool to be configured for the EmptyTarget since it requires valid levels to work on.  Either restore the default back to VectorImage or you'll need to make the necessary changes to allow Plastic tool to work on sub xsheets.

Not sure what else this might be impacting without reviewing every tool.